src/index.ts

```diff
@@ -75,76 +76,66 @@ export function ruleFactory<
     const getReturnValue = () => results.lastValue;
 
     const startTime = performance.now();
-    if (trace) logTrace({ operation: 'begin', startTime });
+    logTrace({ operation: 'begin', startTime });
 
     const parser = init(ruleExpressionLanguage, (term: string) => {
-      if (typeof term === 'string') {
-        try {
-          const result =
-            extractValueOrLiteral(
-              input,
-              term,
-              stepRow,
-              stepCount,
-              ignoreMissingKeys
-            ) ?? get(input, term, undefined as any);
-          // console.log(`TERM: ${term} => ${result}`);
-          if (trace) {
-            logTrace({
-              operation: 'key.lookup',
-              key: term,
-              value: result,
-              stepRow,
-              stepCount,
-            });
-          }
-          return result;
-        } catch (error) {
-          if (trace) {
-            logTrace({
-              operation: 'error',
-              error,
-              rule: term,
-              stepRow,
-              stepCount,
-            });
-          }
-        }
-      } else {
-        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-        throw new Error(`Invalid term: ${term}`);
+      if (typeof term !== 'string') throw new Error(`Invalid term: ${term}`);
```

Comment:
To fix this, we could make the term optional - though maybe not what we want.
```ts
    const parser = init(ruleExpressionLanguage, (term?: string) => {
```
Or, add a bit more defensive posture, by checking the length:
```ts
      if (typeof term !== 'string' || term.length < 1) throw new Error(`Invalid term: ${term}`);
```